services/src/main/groovy/org/jfrog/artifactory/client/impl/ReplicationsImpl.groovy

```diff
@@ -0,0 +1,156 @@
+package org.jfrog.artifactory.client.impl
+
+import groovyx.net.http.ContentType
+import groovyx.net.http.HttpResponseException
+import net.sf.json.JSONArray
+import net.sf.json.JSONObject
+import org.jfrog.artifactory.client.Replications
+import org.jfrog.artifactory.client.model.LocalReplication
+import org.jfrog.artifactory.client.model.Replication
+import org.jfrog.artifactory.client.model.impl.LocalReplicationImpl
+import org.jfrog.artifactory.client.model.impl.RemoteReplicationImpl
+import org.jfrog.artifactory.client.model.impl.RepositoryTypeImpl
+
+class ReplicationsImpl implements Replications {
+
+    private String baseApiPath
+
+    private ArtifactoryImpl artifactory
+
+    private String repoKey
+
+    ReplicationsImpl(ArtifactoryImpl artifactory, String baseApiPath, String repoKey) {
+        this.artifactory = artifactory
+        this.baseApiPath = baseApiPath
+        this.repoKey = repoKey
+    }
+
+    @Override
+    List<Replication> list() {
+        // Determine the type of the repository (not all repository types support replications)
+        def repository = artifactory.repository(repoKey).get()
+
+        if (!repository) {
+            throw new RuntimeException("The repository '${repoKey}' doesn't exist")
+        }
+
+        def path = "${getReplicationsApi()}${repoKey}"
+
+        if ((repository.rclass == RepositoryTypeImpl.LOCAL) || (repository.rclass == RepositoryTypeImpl.REMOTE)) {
```

Comment:
The reason why I wrote "(repository.rclass == RepositoryTypeImpl.LOCAL) || (repository.rclass == RepositoryTypeImpl.REMOTE)" instead of "(repository.rclass != RepositoryTypeImpl.VIRTUAL)" is to cope with possible new repository types in the future.

If JFrog introduces a 4th repository type in the future, the first "complex" condition will require no change. This isn't true with the second condition which implicitly expects that only 3 repository types exist.

I'll do the change nonetheless